src/proxy/IReplicatedMap.ts

```diff
@@ -0,0 +1,169 @@
+import * as Promise from 'bluebird';
+import {DistributedObject} from '../DistributedObject';
+import {Predicate} from '../core/Predicate';
+import {IMapListener} from '../core/MapListener';
+import Long = require('long');
+import {ArrayComparator} from '../util/ArrayComparator';
+
+export interface IReplicatedMap<K, V> extends DistributedObject {
+    /**
+     * Associates a given value to the specified key and replicates it to the
+     * cluster. If there is an old value, it will be replaced by the specified
+     * one and returned from the call.
+     *
+     * @param key key with which the specified value is to be associated.
+     * @param value value to be associated with the specified key.
+     * @param ttl milliseconds to be associated with the specified key-value pair.
+     * @return old value if there was any, `null` otherwise.
+     */
+    put(key: K, value: V, ttl: Long|number|string): Promise<V>;
```

Comment:
"long" library provides maintenance of accepting strings - Long.fromValue accepts both strings and numbers, and keep in mind that `ttl` by spec should be int64, but js supports int32 only, thus strings are the rescue to support big numbers.